tests/analog/test_patterns.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+import pytest
+import torch
+from metrics import ATOL_DICT
+
+from qadence import (
+    AnalogRX,
+    BackendName,
+    DifferentiableBackend,
+    DiffMode,
+    Parameter,
+    QuantumCircuit,
+    QuantumModel,
+    total_magnetization,
+)
+from qadence.analog.addressing import AddressingPattern
+from qadence.analog.interaction import add_interaction
+from qadence.backends.pulser.backend import Backend as PulserBackend
+from qadence.backends.pulser.config import Configuration
+from qadence.backends.pyqtorch.backend import Backend as PyqBackend
+
+
+@pytest.mark.parametrize(
+    "max_amp,max_det",
+    [(0.0, 10.0), (15.0, 0.0), (15.0, 9.0)],
+)
+@pytest.mark.parametrize(
+    "spacing",
+    [8.0, 30.0],
+)
+def test_pulser_pyq_addressing(max_amp: float, max_det: float, spacing: float) -> None:
+    n_qubits = 3
+    block = AnalogRX("x")
```

Comment:
Can you please use a more complicated circuit with at least all the analog rotations?

Reply:
sure